### PR TITLE
fix: make timeline follow table layout

### PR DESCRIPTION
**Pull request by sashuk**
[FX-3801]

### Description

This pull request sets the proper layout of Timeline component, so it keeps the constant width of cells in the same column (table layout)

Design confirmation https://toptal-core.slack.com/archives/CCC3GP6CC/p1698140396833409

### How to test

- [Temploy](https://picasso.toptal.net/fx-3801-fix-timeline-row-s-date-column-to-have-the-same-width-for-all-kind-of-dates)
- All checks should pass
- Compare the timeline in https://staff-portal-pr-11286.toptal.rocks/referrals (https://github.com/toptal/staff-portal/pull/11286) and https://staff-portal.toptal.net/referrals pages
- Check the visual test changes in https://happo.io/a/675/p/1189/compare/7557289dc4e7a2806505560821997b7db136c712/6ef4ceea8ad9479b6d3f6e3fd879fc59585133ce
- Additionally tested in Firefox and Safari

Screenshot 2023-10-24 at 15 12 09

Screenshot 2023-10-24 at 15 12 32

### Screenshots

Before

Screenshot 2023-10-23 at 15 36 54

After

Screenshot 2023-10-24 at 11 10 47

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3801]: https://toptal-core.atlassian.net/browse/FX-3801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

File: packages/picasso/src/Timeline/styles.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { createStyles } from '@material-ui/core/styles'
+
+export default () =>
+  createStyles({
+    root: {
+      display: 'table',
+      height: 'fit-content',
```
Comment (sashuk) [inline packages/picasso/src/Timeline/styles.ts:7]: Usage details https://caniuse.com/?search=width%3A%20fit-content

File: packages/picasso/src/TimelineRow/styles.ts [hunk 2]
```diff
@@ -47,4 +47,14 @@ export default ({ palette }: Theme) =>
       width: 0,
       borderLeft: `1px dashed ${palette.grey.main2}`,
     },
+    tableRow: {
+      display: 'table-row',
+    },
+    tableCell: {
+      display: 'table-cell',
+      verticalAlign: 'top',
```
Comment (sashuk) [inline packages/picasso/src/TimelineRow/styles.ts:55]: Needed to align elements according to BASE

File: packages/picasso/src/TimelineRow/styles.ts [hunk 3]
```diff
@@ -47,4 +47,14 @@ export default ({ palette }: Theme) =>
...
+    },
+    tableCellContent: {
+      height: '100%',
```
Comment (sashuk) [inline packages/picasso/src/TimelineRow/styles.ts:58]: Needed to show the connector (when needed)

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Comment by sashuk** [comment]
@mkrl Great idea, I'll extend existing example a bit with uneven width, thank you!

**Review by TomasSlama — APPROVED** [review]

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Pull request merged**